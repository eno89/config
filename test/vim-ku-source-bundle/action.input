" Test: vim-ku-source-bundle action
runtime! plugin/ku.vim
runtime! plugin/ku/kind/buffer.vim plugin/ku/kind/file.vim
runtime! plugin/ku/source/bundle.vim
runtime! plugin/bundle.vim
call vspec#hint({})
function s:describe__ku_action_bundle_args__changed_and_defined()  "{{{1
  It requires the following initialization

  Should argv() ==# []
  Should bufexists('a') == 0
  Should bufexists('b') == 0
  Should bufexists('c') == 0
  Should bufexists('foo') == 0

  let g:bundle_dictionary = {'test': ['a', 'b', 'c']}
  Should bundle#available_bundles() ==# ['test']
  Should bundle#files('test') ==# ['a', 'b', 'c']

  silent edit foo
  normal! Alice
  let bufnr = bufnr('')
  Should bufname('') ==# 'foo'
  Should &l:modified != 0

  It should fail

  silent let _ = ku#source#bundle#action_args({'word': 'test'})
  Should _ == 0
  Should argv() ==# []
  Should bufnr('') == bufnr

  It requires to reset side effect

  bwipeout! foo
  silent! argdo bwipeout
  silent! argdelete *
  Should argv() ==# []
  Should bufexists('a') == 0
  Should bufexists('b') == 0
  Should bufexists('c') == 0
  Should bufexists('foo') == 0

  let g:bundle_dictionary = {}
  Should bundle#available_bundles() ==# []
  Should bundle#files('test') ==# []
endfunction




function s:describe__ku_action_bundle_args__changed_and_undefined()  "{{{1
  It requires the following initialization

  Should argv() ==# []
  Should bufexists('a') == 0
  Should bufexists('b') == 0
  Should bufexists('c') == 0
  Should bufexists('foo') == 0

  let g:bundle_dictionary = {'test': ['a', 'b', 'c']}
  Should bundle#available_bundles() ==# ['test']
  Should bundle#files('test') ==# ['a', 'b', 'c']

  silent edit foo
  normal! Alice
  let bufnr = bufnr('')
  Should bufname('') ==# 'foo'
  Should &l:modified != 0

  It should fail

  silent let _ = ku#source#bundle#action_args({'word': 'test??'})
  Should type(_) == type('')
  Should argv() ==# []
  Should bufnr('') == bufnr

  It requires to reset side effect

  bwipeout! foo
  silent! argdo bwipeout
  silent! argdelete *
  Should argv() ==# []
  Should bufexists('a') == 0
  Should bufexists('b') == 0
  Should bufexists('c') == 0
  Should bufexists('foo') == 0

  let g:bundle_dictionary = {}
  Should bundle#available_bundles() ==# []
  Should bundle#files('test') ==# []
endfunction




function s:describe__ku_action_bundle_args__unchanged_and_defined()  "{{{1
  It requires the following initialization

  Should argv() ==# []
  Should bufexists('a') == 0
  Should bufexists('b') == 0
  Should bufexists('c') == 0
  Should bufexists('foo') == 0

  let g:bundle_dictionary = {'test': ['a', 'b', 'c']}
  Should bundle#available_bundles() ==# ['test']
  Should bundle#files('test') ==# ['a', 'b', 'c']

  silent edit foo
  let bufnr = bufnr('')
  Should bufname('') ==# 'foo'
  Should &l:modified == 0

  It should succeed

  silent let _ = ku#source#bundle#action_args({'word': 'test'})
  Should _ == 0
  Should argv() ==# ['a', 'b', 'c']
  Should argidx() == 0
  Should bufnr('') != bufnr
  Should bufname('') ==# 'a'

  It requires to reset side effect

  bwipeout! foo
  silent! argdo bwipeout
  silent! argdelete *
  Should argv() ==# []
  Should bufexists('a') == 0
  Should bufexists('b') == 0
  Should bufexists('c') == 0
  Should bufexists('foo') == 0

  let g:bundle_dictionary = {}
  Should bundle#available_bundles() ==# []
  Should bundle#files('test') ==# []
endfunction




function s:describe__ku_action_bundle_args__unchanged_and_undefined()  "{{{1
  It requires the following initialization

  Should argv() ==# []
  Should bufexists('a') == 0
  Should bufexists('b') == 0
  Should bufexists('c') == 0
  Should bufexists('foo') == 0

  let g:bundle_dictionary = {'test': ['a', 'b', 'c']}
  Should bundle#available_bundles() ==# ['test']
  Should bundle#files('test') ==# ['a', 'b', 'c']

  silent edit foo
  let bufnr = bufnr('')
  Should bufname('') ==# 'foo'
  Should &l:modified == 0

  It should fail

  silent let _ = ku#source#bundle#action_args({'word': 'test??'})
  Should type(_) == type('')
  Should argv() ==# []
  Should bufnr('') == bufnr

  It requires to reset side effect

  bwipeout! foo
  silent! argdo bwipeout
  silent! argdelete *
  Should argv() ==# []
  Should bufexists('a') == 0
  Should bufexists('b') == 0
  Should bufexists('c') == 0
  Should bufexists('foo') == 0

  let g:bundle_dictionary = {}
  Should bundle#available_bundles() ==# []
  Should bundle#files('test') ==# []
endfunction




function s:describe__ku_action_bundle_args_x__changed_and_defined()  "{{{1
  It requires the following initialization

  Should argv() ==# []
  Should bufexists('a') == 0
  Should bufexists('b') == 0
  Should bufexists('c') == 0
  Should bufexists('foo') == 0

  let g:bundle_dictionary = {'test': ['a', 'b', 'c']}
  Should bundle#available_bundles() ==# ['test']
  Should bundle#files('test') ==# ['a', 'b', 'c']

  silent edit foo
  normal! Alice
  let bufnr = bufnr('')
  Should bufname('') ==# 'foo'
  Should &l:modified != 0

  It should succeed

  silent let _ = ku#source#bundle#action_args_x({'word': 'test'})
  Should _ == 0
  Should argv() ==# ['a', 'b', 'c']
  Should argidx() == 0
  Should bufnr('') != bufnr
  Should bufname('') ==# 'a'

  It requires to reset side effect

  bwipeout! foo
  silent! argdo bwipeout
  silent! argdelete *
  Should argv() ==# []
  Should bufexists('a') == 0
  Should bufexists('b') == 0
  Should bufexists('c') == 0
  Should bufexists('foo') == 0

  let g:bundle_dictionary = {}
  Should bundle#available_bundles() ==# []
  Should bundle#files('test') ==# []
endfunction




function s:describe__ku_action_bundle_args_x__changed_and_undefined()  "{{{1
  It requires the following initialization

  Should argv() ==# []
  Should bufexists('a') == 0
  Should bufexists('b') == 0
  Should bufexists('c') == 0
  Should bufexists('foo') == 0

  let g:bundle_dictionary = {'test': ['a', 'b', 'c']}
  Should bundle#available_bundles() ==# ['test']
  Should bundle#files('test') ==# ['a', 'b', 'c']

  silent edit foo
  normal! Alice
  let bufnr = bufnr('')
  Should bufname('') ==# 'foo'
  Should &l:modified != 0

  It should fail

  silent let _ = ku#source#bundle#action_args_x({'word': 'test??'})
  Should type(_) == type('')
  Should argv() ==# []
  Should bufnr('') == bufnr

  It requires to reset side effect

  bwipeout! foo
  silent! argdo bwipeout
  silent! argdelete *
  Should argv() ==# []
  Should bufexists('a') == 0
  Should bufexists('b') == 0
  Should bufexists('c') == 0
  Should bufexists('foo') == 0

  let g:bundle_dictionary = {}
  Should bundle#available_bundles() ==# []
  Should bundle#files('test') ==# []
endfunction




function s:describe__ku_action_bundle_args_x__unchanged_and_defined()  "{{{1
  It requires the following initialization

  Should argv() ==# []
  Should bufexists('a') == 0
  Should bufexists('b') == 0
  Should bufexists('c') == 0
  Should bufexists('foo') == 0

  let g:bundle_dictionary = {'test': ['a', 'b', 'c']}
  Should bundle#available_bundles() ==# ['test']
  Should bundle#files('test') ==# ['a', 'b', 'c']

  silent edit foo
  let bufnr = bufnr('')
  Should bufname('') ==# 'foo'
  Should &l:modified == 0

  It should succeed

  silent let _ = ku#source#bundle#action_args_x({'word': 'test'})
  Should _ == 0
  Should argv() ==# ['a', 'b', 'c']
  Should argidx() == 0
  Should bufnr('') != bufnr
  Should bufname('') ==# 'a'

  It requires to reset side effect

  bwipeout! foo
  silent! argdo bwipeout
  silent! argdelete *
  Should argv() ==# []
  Should bufexists('a') == 0
  Should bufexists('b') == 0
  Should bufexists('c') == 0
  Should bufexists('foo') == 0

  let g:bundle_dictionary = {}
  Should bundle#available_bundles() ==# []
  Should bundle#files('test') ==# []
endfunction




function s:describe__ku_action_bundle_args_x__unchanged_and_undefined()  "{{{1
  It requires the following initialization

  Should argv() ==# []
  Should bufexists('a') == 0
  Should bufexists('b') == 0
  Should bufexists('c') == 0
  Should bufexists('foo') == 0

  let g:bundle_dictionary = {'test': ['a', 'b', 'c']}
  Should bundle#available_bundles() ==# ['test']
  Should bundle#files('test') ==# ['a', 'b', 'c']

  silent edit foo
  let bufnr = bufnr('')
  Should bufname('') ==# 'foo'
  Should &l:modified == 0

  It should fail

  silent let _ = ku#source#bundle#action_args_x({'word': 'test??'})
  Should type(_) == type('')
  Should argv() ==# []
  Should bufnr('') == bufnr

  It requires to reset side effect

  bwipeout! foo
  silent! argdo bwipeout
  silent! argdelete *
  Should argv() ==# []
  Should bufexists('a') == 0
  Should bufexists('b') == 0
  Should bufexists('c') == 0
  Should bufexists('foo') == 0

  let g:bundle_dictionary = {}
  Should bundle#available_bundles() ==# []
  Should bundle#files('test') ==# []
endfunction




function s:describe__ku_action_bundle_load()  "{{{1
endfunction




function s:describe__ku_action_bundle_load_x()  "{{{1
endfunction




" __END__  "{{{1
" vim: filetype=vim foldmethod=marker
