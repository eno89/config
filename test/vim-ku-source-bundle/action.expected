==== ku_action_bundle_args changed and defined
---- It requires the following initialization 
.........
---- It should fail 
...
---- It requires to reset side effect 
.......

==== ku_action_bundle_args changed and undefined
---- It requires the following initialization 
.........
---- It should fail 
...
---- It requires to reset side effect 
.......

==== ku_action_bundle_args unchanged and defined
---- It requires the following initialization 
.........
---- It should succeed 
.....
---- It requires to reset side effect 
.......

==== ku_action_bundle_args unchanged and undefined
---- It requires the following initialization 
.........
---- It should fail 
...
---- It requires to reset side effect 
.......

==== ku_action_bundle_args_x changed and defined
---- It requires the following initialization 
.........
---- It should succeed 
.....
---- It requires to reset side effect 
.......

==== ku_action_bundle_args_x changed and undefined
---- It requires the following initialization 
.........
---- It should fail 
...
---- It requires to reset side effect 
.......

==== ku_action_bundle_args_x unchanged and defined
---- It requires the following initialization 
.........
---- It should succeed 
.....
---- It requires to reset side effect 
.......

==== ku_action_bundle_args_x unchanged and undefined
---- It requires the following initialization 
.........
---- It should fail 
...
---- It requires to reset side effect 
.......

==== ku_action_bundle_load 

==== ku_action_bundle_load_x 



**** Result ****


158 examples, 0 failures
